cocotb/scheduler.py

```diff
@@ -665,7 +670,12 @@ def _trigger_from_any(self, result):
         if isinstance(result, cocotb.triggers.Waitable):
             return self._trigger_from_waitable(result)
 
-        raise TypeError
+        raise TypeError(
+            "Coroutine yielded something the scheduler can't handle\n"
+            "Got type: {} repr: {!r} str: {}\n"
+            "Did you forget to decorate with @cocotb.coroutine?"
+            .format(type(result), result, result)
```

Comment:
Can you use the exact same message as before, only replacing the first sentence? (Either one is fine.)

Reply:
Isn't this already the same message as before, or am I missing something?

Reply:
Sorry for the confusion, I meant "before" as in line scheduler.py:609-611 (the messages are similar in content, yet written in a slightly different way).

Reply:
Done